samples/test/metrics_visualization_v2_test.py

```diff
@@ -12,15 +12,180 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import unittest
+import unittest.mock as mock
+from pprint import pprint
 import kfp
+import kfp_server_api
+
 from .metrics_visualization_v2 import metrics_visualization_pipeline
-from .util import run_pipeline_func, TestCase
+from .util import run_pipeline_func, TestCase, KfpMlmdClient
+
+
+def verify(
+    run: kfp_server_api.ApiRun, mlmd_connection_config, argo_workflow_name: str,
+    **kwargs
+):
+    t = unittest.TestCase()
+    t.maxDiff = None  # we always want to see full diff
+    t.assertEqual(run.status, 'Succeeded')
+    client = KfpMlmdClient(mlmd_connection_config=mlmd_connection_config)
+    tasks = client.get_tasks(argo_workflow_name=argo_workflow_name)
+
+    task_names = [*tasks.keys()]
+    t.assertCountEqual(
+        task_names,
+        ['wine-classification', 'iris-sgdclassifier', 'digit-classification'],
+        'task names'
+    )
+
+    wine_classification: KfpTask = tasks.get('wine-classification')
+    iris_sgdclassifier: KfpTask = tasks.get('iris-sgdclassifier')
+    digit_classification: KfpTask = tasks.get('digit-classification')
+
+    pprint('======= wine classification task =======')
+    pprint(wine_classification.get_dict())
+    pprint('======= iris sgdclassifier task =======')
+    pprint(iris_sgdclassifier.get_dict())
+    pprint('======= digit classification task =======')
+    pprint(digit_classification.get_dict())
 
```

Comment:
NIT: Is it used for debugging? I assume that `t.assertEqual()` will print out the same content if the assertion failed. So these seems not necessary to print them here.

Reply:
Yes, these are for debugging.
When I tried, I recall that `t.assertEqual` doesn't print out everything, it will only print out the diff.
Therefore, I preferred logging the entire details all the time.

Reply:
I see, sounds good on keeping the logging then.